front-end/src/components/main-page/OverrideModal.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import Modal from 'react-modal';
+import BaseButton from '../general/BaseButton';
+import PropTypes from 'prop-types';
+
+class OverrideModal extends React.Component {
+
+    static propTypes = {
+        show: PropTypes.bool.isRequired,
+        toggleModal: PropTypes.func.isRequired,
+        uploadFileWithOverride: PropTypes.func.isRequired,
+    }
```

Comment:
Nit: This hasn't been how we are defining our proptypes. This is definitely a cool implementation, but I think it is better we stay consistent.